storage/object/how-to/restore-an-object-from-glacier.mdx

```diff
@@ -57,22 +57,29 @@ categories:
 
 ## How to restore all objects in a bucket
 
-If you have numerous files in a bucket that you would like to restore, it may be more convenient to use the command line, for example with the [s3cmd tool](/tutorials/s3cmd/). To restore all files recursively in a bucket, you can invoke this command:
+If you have numerous files in a bucket that you would like to restore, we recommen using the command line interface with for example with a dedicated tool such as the [AWS CLI](/storage/object/api-cli/object-storage-aws-cli/).
 
-```
-s3cmd restore -v --recursive s3://<bucket-name>
-```
+1. Run the command below in a terminal to create a list of objects to restore and store it as a text file. Make sure to replace `<YOUR-BUCKET>` by the name of your bucket.
 
+    ```sh
+    aws s3api list-objects-v2 --bucket <YOUR-BUCKET> --query "Contents[?StorageClass=='GLACIER']" --output text | awk '{print $2}' > glacier-restore.txt
+    ```
+    <Message type="note">
+    The `list-objects-v2` operation is [limited to 1,000 objects](https://docs.aws.amazon.com/cli/latest/reference/s3api/list-objects-v2.html#description).
+    </Message>
+    
+2. Run the following command to restore every object listed at the previous step. Make sure to replace `<YOUR-BUCKET>` by the name of your bucket, and `NUM` by the desired number of days.
+
+    ```sh
+    for x in `cat glacier-restore.txt`
+      do
+        aws s3api restore-object --restore-request Days=NUM --bucket <YOUR-BUCKET> --key "$x"
+      done
+    ```
 <Message type="note">
-  The `s3cmd restore` command does not support permanent restores, and only restores objects for one day by default. Use the `--restore-days=NUM` parameter to specify the desired number of days.
+  The `s3api restore-object` command does not support permanent restores, and only restores objects for one day by default. Use the `Days=NUM` parameter to specify the desired number of days.
 </Message>
 
-To count the number of files still in `Glacier` (see [restore time](#restore-time) below, this is not immediate), you can use:
-
-```
-s3cmd ls -l --recursive s3://<bucket-name> |grep GLACIER|wc -l
-```
-
 ## Restore time
 
 The time it takes to restore an object depends on the size of the object and on if [multipart](/storage/object/concepts/#multipart-uploads) is configured. If your object is larger than 1 MB, it can take anywhere from a few minutes to 24 hours for restore to start.
```

Comment:
```suggestion
The time it takes to restore an object depends on the size of the object and if [multipart](/storage/object/concepts/#multipart-uploads) is configured. If your object is larger than 1 MB, it can take anywhere from a few minutes to 24 hours for restore to start.
```